### DISCUSSION TITLE
Radio button group inside boxes

### DISCUSSION THREAD

**Discussion by ghost** [chakra-ui/chakra-ui#2889] [Q&A] [answered]
I want radio group functionality but with buttons instead of the typical radio button.

I couldn't figure out a way to do that with actual buttons, so I wrapped each radio button inside a box.

It LOOKS great, imho. 

![Chakra Radio](https://user-images.githubusercontent.com/55907892/103165395-939b4f00-47d4-11eb-80d8-4bafe6d88d1c.png)

Unfortunately, (and, of course!), only the radio button and the caption receive the click event.

I'd like it if clicking anywhere in a colored box selected the radio button.

Is there a way to do this?

**Comment by with-heart** [comment]
Could you put together a CodeSandbox for this? It'd make it easier to play with and help you out!

**Reply by ghost** [reply]
Thanks for that. Sorry for the delayed response. Been a crazy day. 

I've not done a CodeSandbox before, so it may take me a bit to figure out. I'll work on both tonight.

**Reply by chr-ge** [reply]
You can work off this one: https://codesandbox.io/s/chakra-ui-javascript-lzzg9

**Reply by ghost** [reply]
That was a HUGE help!

Here's a basic version:

https://codesandbox.io/s/small-platform-9xukc?file=/src/index.js

Basically, if the User clicks anywhere in the box, I want it to be the same as if they clicked on the radio button.

Does that help?

**Comment by ghost** [comment]
I created the sandbox. Any answers on how to do this?

**Comment by chr-ge** [comment, marked as answer]
Adding `w='100%'` and `h='100%'` to each radio should work well. https://codesandbox.io/s/vigorous-dream-hukz1?file=/src/index.js

**Reply by ghost** [reply]
Awesome! Thanks so much!!
